src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -114,7 +114,13 @@ bool ChaseMovementGenerator<T>::DoUpdate(T* owner, uint32 time_diff)
     if (PositionOkay(owner, target, maxRange, angle) && !owner->HasUnitState(UNIT_STATE_CHASE_MOVE))
         return true;
 
-    bool moveToward = !owner->IsInDist(target, maxRange);
+    float tarX, tarY, tarZ;
+    target->GetPosition(tarX, tarY, tarZ);
+
+    bool withinRange = owner->IsInDist(target, maxRange);
+    bool withinLOS   = owner->IsWithinLOS(tarX, tarY, tarZ);
+    bool moveToward  = !withinRange || (withinRange && !withinLOS);
```

Comment:
`bool moveToward  = !(withinRange && withinLOS);`

Reply:
It's more terse, but it took me a moment to parse and unpack the different cases. The other one spells it all out which, in my opinion, makes it easier to read. So I'm not sure what the gain here would be since these things get optimized by the compiler anyway. Any thoughts on this?

Reply:
Just readbility.

Reply:
Alright, let's not get stuck on preferences then :-)